src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cloud.azure.management;
+
+import com.microsoft.windowsazure.Configuration;
+import com.microsoft.windowsazure.core.utils.KeyStoreType;
+import com.microsoft.windowsazure.management.compute.ComputeManagementClient;
+import com.microsoft.windowsazure.management.compute.ComputeManagementService;
+import com.microsoft.windowsazure.management.compute.models.HostedServiceGetDetailedResponse;
+import com.microsoft.windowsazure.management.configuration.ManagementConfiguration;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.cloud.azure.AzureServiceDisableException;
+import org.elasticsearch.cloud.azure.AzureServiceRemoteException;
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.SettingsFilter;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.elasticsearch.cloud.azure.management.AzureComputeService.Management.*;
+
+/**
+ *
+ */
+public class AzureComputeServiceImpl extends AbstractLifecycleComponent<AzureComputeServiceImpl>
+    implements AzureComputeService {
+
+    static final class Azure {
+        private static final String ENDPOINT = "https://management.core.windows.net/";
+    }
+
+    private final ComputeManagementClient computeManagementClient;
+    private final String serviceName;
+
+    @Inject
+    public AzureComputeServiceImpl(Settings settings, SettingsFilter settingsFilter) {
+        super(settings);
+        // Cloud management API settings we need to hide
+        settingsFilter.addFilter(KEYSTORE_PATH);
+        settingsFilter.addFilter(KEYSTORE_PASSWORD);
+        settingsFilter.addFilter(KEYSTORE_TYPE);
+        settingsFilter.addFilter(SUBSCRIPTION_ID);
+
+        // Deprecated Cloud management API settings we need to hide
+        // TODO Remove in 3.0.0
+        settingsFilter.addFilter(KEYSTORE_DEPRECATED);
+        settingsFilter.addFilter(PASSWORD_DEPRECATED);
+        settingsFilter.addFilter(SUBSCRIPTION_ID_DEPRECATED);
```

Comment:
Does this mean that if somebody temporary disables the service all settings will become exposed?